### ISSUE TITLE
[no-inner-declaration] error inside namespace and module

### ISSUE THREAD

**Issue by ficristo** [typescript-eslint/typescript-eslint#239]
**Repro**

```JSON
{
  "rules": {
    "no-inner-declarations": "error"
  }
}
```

```TS
export namespace Foo {
    export function bar() {}
}

export module Bar {
    export function foo() {}
}
```

**Expected Result**
No errors.

**Actual Result**
2 errors.

**Additional Info**
-

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.3.0` |
| `@typescript-eslint/parser`        | `1.3.0` |
| `TypeScript`                       | `3.3.3` |
| `ESLint`                           | `5.13.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
I would suggest not using this rule.
Unless you're supporting browsers / node versions that don't support ES6 (i.e. like IE6 and older), then the browser should have:
- block scoped variables (via `let` / `const`)
- block scoped functions

This combined with typescript's type checking, which will ensure you're not using a variable defined in a child scope (i.e. an unavailable variable).
If you're worried about shadowing a variable in parent scope, then instead look to using [`no-shadow`](https://eslint.org/docs/rules/no-shadow) instead.

---

The functionality is hard-coded into the base rule, so we'd have to create a base rule extension within this plugin, which is a decent chunk of work.

**Comment by JoshuaKGoldberg** [comment]
Err, I no longer repro the original issue of 2 erroneous errors: https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=KYDwDg9gTgLgBAOwIYFtgGcxIMbDgMQgjgG8AoOOUSWOAMwFcFsYBLCBOAIySgAoAlKTgBfMmLLVo8FBAAmDADZ4AQr1IUq4afSYt2nOkUHCxYoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAS110WkIBNEBjeAQ2gYBdj9dUNILp9pI4MAF8QwoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

Namespaces are also not a feature recommended for organizing TypeScript codebases anymore. So this is really an edge case getting edgier by the year.

Closing as aged away. Thanks!
